archive/walk.go

```diff
@@ -66,3 +73,73 @@ func SpanWalk(ark *Archive, v SpanVisitor) error {
 func RmDirs(ark *Archive) error {
 	return Walk(ark, ark.dataSrc.RemoveAll)
 }
+
+type multiSource struct {
+	ark   *Archive
+	paths []string
+}
+
+// NewMultiSource returns a driver.MultiSource for an Archive. If no paths are
+// specified, the MultiSource will send a source for each chunk file, and
+// report the same ordering as the archive. Otherwise, the sources come from
+// localizing the given paths to each chunk's directory (recognizing "_" as the
+// chunk file itself), with no defined ordering.
+func NewMultiSource(ark *Archive, paths []string) driver.MultiSource {
```

Comment:
Now I see driver.MultiSource used here, it seems like an inversion to have a core library use and return types defined in an outer package like `driver`. Maybe driver.MultiSource could move to another core library like scanner/. Either way, this doesn't have to be addressed here.